platforms/core-runtime/logging/src/testFixtures/groovy/org/gradle/internal/logging/text/TestStyledTextOutputFactory.java

```diff
@@ -20,34 +20,72 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
+/**
+ * Test fixture for {@link StyledTextOutputFactory} that tracks the styled text produced.
+ *
+ * This is intended to be used in tests and only in cases where a single category/log level is used at a time.
+ *
+ * This fixture allows tests to assert that the correct log category, log level and styled output has been produced.
+ */
 public class TestStyledTextOutputFactory extends AbstractStyledTextOutputFactory implements StyledTextOutputFactory {
     private final List<StyledTextOutput> textOutputs = new ArrayList<StyledTextOutput>();
+    private String category;
+    private LogLevel logLevel;
 
+    /**
+     * {@inheritDoc}
+     *
+     * @implNote This implementation tracks the output of the most recently created category and log level. If the category or log level
+     * changes, the tracked output is reset.
```

Comment:
💭 I imagine this usage note will be forgotten/unread by people, who will use this fixture in a test where the category/log level changes, and then waste time debugging the test to see what happened.

I think it would be better to just throw an exception if the tracked category/level changes saying this fixture won't work here.

Reply:
Alternately, you could rename this type to `TestSingleStyledTextOutputFactory` or something, and throw immediately upon calling create more than once.  I think that might be even clearer.

Reply:
done